### PR TITLE
Freight enhancement and validation script

### PR DESCRIPTION
**Pull request by haitamlaarabi**
This change is [Reviewable](https://reviewable.io/reviews/LBNL-UCB-STI/beam/3825)

### CODE REVIEW

**Comment by haitamlaarabi** [comment]
Test!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
Test now!

**Comment by haitamlaarabi** [comment]
test!

**Comment by haitamlaarabi** [comment]
test!

File: src/main/resources/beam-template.conf [hunk 1]
```diff
@@ -498,7 +498,7 @@ beam.physsim.ptSampleSize = "double | 1.0"
 beam.physsim.eventsForFullVersionOfVia = true
 beam.physsim.eventsSampling = "double | 1.0"
 beam.physsim.initializeRouterWithFreeFlowTimes = true
-beam.physsim.quick_fix_minCarSpeedInMetersPerSecond = "double | 0.5"
+beam.physsim.minCarSpeedInMetersPerSecond = "double | 0.5"
```
Comment (dimaopen) [inline src/main/resources/beam-template.conf]: We need to find and replace all the entries of this changed param in the whole project (including rst files)

File: src/main/java/beam/physsim/jdeqsim/cacc/sim/JDEQSimulation.java [hunk 2]
```diff
@@ -87,21 +87,20 @@ private void logInitializeVehiclesOutcome(List<String> vehicleNotFound, Map<Stri
...
         Scheduler scheduler = getScheduler();
         allRoads.clear();
+        double minimumRoadSpeedInMetersPerSecond = beamConfig.beam().physsim().minCarSpeedInMetersPerSecond();
         if (maybeCaccSettings.nonEmpty()) {
             CACCSettings caccSettings = maybeCaccSettings.get();
             for (Link link : scenario.getNetwork().getLinks().values()) {
                 allRoads.put(link.getId(), new Road(scheduler, link,
                         caccSettings.speedAdjustmentFactor(),
-                        caccSettings.adjustedMinimumRoadSpeedInMetersPerSecond(),
+                        minimumRoadSpeedInMetersPerSecond,
```
Comment (dimaopen) [inline src/main/java/beam/physsim/jdeqsim/cacc/sim/JDEQSimulation.java:96]: If we don't use `adjustedMinimumRoadSpeedInMetersPerSecond` anymore then we need to remove it from the project completely (including documentation)

File: src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala [hunk 3]
```diff
@@ -258,6 +262,8 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with Stash with Expon
           eventBuilderActor,
           beamServices.beamCustomizationAPI.beamVehicleAfterUseFuelHook
         )
+      if (this.currentBeamVehicle.isFreightVehicle)
+        log.info(s"RouteE: Vehicle ${currentBeamVehicle.id} - fuelConsumed: ${fuelConsumed}")
```
Comment (dimaopen) [inline src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala]: Shouldn't these 3 log entries go to debug instead of info?

**Review by dimaopen — CHANGES_REQUESTED** [review]
Looks good except a few requests

File: src/main/java/beam/physsim/jdeqsim/cacc/sim/JDEQSimulation.java (hunk 2 above)
Comment (haitamlaarabi) [reply src/main/java/beam/physsim/jdeqsim/cacc/sim/JDEQSimulation.java:96]: deleted all occurrence and I mentioned this in the template file:
```
beam.physsim.minCarSpeedInMetersPerSecond = "double | 0.5" # heads up, this affects beam.physsim.jdeqsim.cacc
# The following parameter has been replaced by beam.physsim.minCarSpeedInMetersPerSecond = 2.0 
# beam.physsim.jdeqsim.cacc.adjustedMinimumRoadSpeedInMetersPerSecond = "double | 1.3"
```

File: src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala (hunk 3 above)
Comment (haitamlaarabi) [reply src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala]: it was a temporary debug log. I'm removing it.

**Comment by haitamlaarabi** [comment]
Test!

File: src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala [hunk 4]
```diff
@@ -247,7 +247,11 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with Stash with Expon
       val currentVehicleUnderControl = data.currentVehicle.headOption
         .getOrElse(throw new RuntimeException("Current Vehicle is not available."))
       val isLastLeg = data.currentLegPassengerScheduleIndex + 1 == data.passengerSchedule.schedule.size
+      if (this.currentBeamVehicle.isFreightVehicle)
+        log.info(s"RouteE: Vehicle ${currentBeamVehicle.id} - vehicleType ${currentBeamVehicle.beamVehicleType.id}")
```
Comment (dimaopen) [inline src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala]: This entry should probably be removed too

File: src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala [hunk 5]
```diff
@@ -247,7 +247,11 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with Stash with Expon
...
       val payloadInKg = payloadInKgForLeg(currentLeg, data)
+      if (this.currentBeamVehicle.isFreightVehicle)
+        log.info(s"RouteE: Vehicle ${currentBeamVehicle.id} - payloadInKg: ${payloadInKg}")
```
Comment (dimaopen) [inline src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala]: This entry should probably be removed too

**Review by dimaopen — APPROVED** [review]
Looks good. Though these params:
`adjustedMinimumRoadSpeedInMetersPerSecond`
`quick_fix_minCarSpeedInMetersPerSecond`
are still in the production config files. May be we cannot remove them because of the old beam code that can be used with production.
Also DrivesVehicle still contains some log.info that can pollute the log file.

**Comment by haitamlaarabi** [comment]
> adjustedMinimumRoadSpeedInMetersPerSecond

I think default value of quick_fix_minCarSpeedInMetersPerSecond is fine, since it wasn't solving the problem it was supposed to solve anyway.

**Comment by haitamlaarabi** [comment]
test!

**Pull request merged**